### PR TITLE
RavenDB-20464 Move Admin JS Console view to react

### PR DESCRIPTION
**Pull request by kwiato**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-20464/Move-Admin-JS-Console-view-to-react

### Additional description

AceEditor in react, styles for AceEditor in scss, Admin JS Console view in react and with new look

### Type of change

- New feature

### How risky is the change?

- Low 

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing by Contributor

- It has been verified by manual testing

### Testing by RavenDB QA team

- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Studio/typescript/components/common/AceEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                    commands={commands}
+                    {...rest}
+                />
+                <span className="fullScreenModeLabel">Press Shift+F11 to enter full screen mode</span>
+            </div>
+            {errorMessage && (
+                <div className="text-danger small" style={{ backgroundColor: "#5e3746", padding: "4px" }}>
+                    {errorMessage}
+                </div>
+            )}
```
Comment (kalczur) [inline src/Raven.Studio/typescript/components/common/AceEditor.tsx]: we leave it like this?

File: src/Raven.Studio/typescript/components/pages/resources/manageServer/adminJsConsole/AdminJsConsole.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { ComponentMeta, ComponentStory } from "@storybook/react";
+import { withStorybookContexts, withBootstrap5 } from "test/storybookTestUtils";
+import AdminJsConsole from "./AdminJsConsole";
+
+export default {
+    title: "Pages/ManageServer",
```
Comment (kalczur) [inline src/Raven.Studio/typescript/components/pages/resources/manageServer/adminJsConsole/AdminJsConsole.stories.tsx]: AdminJsConsole

**Review by kalczur — CHANGES_REQUESTED** [review]

File: src/Raven.Studio/typescript/common/autoComplete/remoteMetadataProvider.ts [hunk 3]
```diff
@@ -49,7 +49,7 @@ class remoteMetadataProvider implements queryCompleterProviders {
     }
 
     indexNames(callback: (indexNames: string[]) => void): void {
-        callback(this.indexes());
+        callback(ko.isObservable(this.indexes) ? this.indexes() : this.indexes);
```
Comment (ml054) [inline src/Raven.Studio/typescript/common/autoComplete/remoteMetadataProvider.ts]: you can use ko.unwrap.

File: src/Raven.Studio/typescript/common/rqlLanguageService.ts [hunk 4]
```diff
@@ -16,7 +17,7 @@ class rqlLanguageService {
     
     constructor(
         db: database, 
-        indexes: KnockoutObservable<string[]>,
+        indexes: KnockoutObservable<string[]> | string[],
```
Comment (ml054) [inline src/Raven.Studio/typescript/common/rqlLanguageService.ts]: what if we put here: `indexes: ()=> string[]` ?

Comment (ml054) [reply src/Raven.Studio/typescript/common/rqlLanguageService.ts]: which is "provider" of index names.

File: src/Raven.Studio/typescript/components/common/AceEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+export interface AceEditorProps extends IAceEditorProps {
+    mode: AceEditorMode;
+    languageService?: LanguageService;
+    validationErrorMessage?: string;
+    execute?: (...args: any) => any;
+    setIsValid?: (isValid: boolean) => void;
+}
+
+export default function AceEditor(props: AceEditorProps) {
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/AceEditor.tsx:17]: Did you test if it works with RQL / autocomplete/ JSON etc? We had a lot of customizations in ace editor.

File: src/Raven.Studio/typescript/components/pages/database/tasks/backups/BackupsPage.tsx [hunk 6]
```diff
@@ -274,13 +275,36 @@ export function BackupsPage(props: BackupsPageProps) {
...
                         </Button>
+                        <FlexGrow />
+                        <AboutView>
+                            <Row>
+                                <Col sm={"auto"}>
+                                    <Icon className="fs-1" icon="backup" color="info" margin="m-0" />
+                                </Col>
+                                <Col>
+                                    <p>
+                                        <strong>Backups</strong> save your data at a specific point in time and allow
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/pages/database/tasks/backups/BackupsPage.tsx:293]: how is it realted to moving js console to react?

**Comment by ml054** [comment]
Useful links opens in same tab. Is it expected?

**Comment by ml054** [comment]
image
left margin is too big?

File: src/Raven.Studio/typescript/components/common/AceEditor.tsx (hunk 1 above)
Comment (kalczur) [reply src/Raven.Studio/typescript/components/common/AceEditor.tsx]: i mean inline style

File: src/Raven.Studio/typescript/components/pages/database/tasks/backups/BackupsPage.tsx (hunk 6 above)
Comment (kwiato) [reply src/Raven.Studio/typescript/components/pages/database/tasks/backups/BackupsPage.tsx:293]: Just wanted to test the component directly in studio not in storybook

File: src/Raven.Studio/typescript/components/common/AceEditor.tsx (hunk 5 above)
Comment (kalczur) [reply src/Raven.Studio/typescript/components/common/AceEditor.tsx:17]: I checked for several languages and RQL autocomplete - everything works

File: src/Raven.Studio/typescript/common/rqlLanguageService.ts (hunk 4 above)
Comment (kalczur) [reply src/Raven.Studio/typescript/common/rqlLanguageService.ts]: All right

**Review by ml054 — APPROVED** [review]

**Pull request merged**